scripts/o3de/o3de/utils.py

```diff
@@ -346,3 +374,102 @@ def remove_gem_duplicates(gems: list) -> list:
             else:
                 new_list[names[gem_name]] = gem
     return new_list
+
+
+def update_keys_and_values_in_dict(existing_values: dict, new_values: list or str, remove_values: list or str,
+                      replace_values: list or str):
+    """
+    Updates values within a dictionary by replacing all values or by appending values in the new_values list and 
+    removing values in the remove_values
+    :param existing_values dict to modify
+    :param new_values list of key=value pairs to add to the existing dictionary 
+    :param remove_values list with keys to remove from the existing dictionary 
+    :param replace_values list with key=value pairs to replace existing dictionary with
+
+    returns updated existing dictionary
+    """
+    if replace_values != None:
+        replace_values = replace_values.split() if isinstance(replace_values, str) else replace_values
+        return dict(entry.split('=') for entry in replace_values if '=' in entry)
+    
+    if new_values:
+        new_values = new_values.split() if isinstance(new_values, str) else new_values
+        new_values = dict(entry.split('=') for entry in new_values if '=' in entry)
+        if new_values:
+            existing_values.update(new_values)
+    
+    if remove_values:
+        remove_values = remove_values.split() if isinstance(remove_values, str) else remove_values
+        [existing_values.pop(key) for key in remove_values]
```

Comment:
Shouldn't we also split the key in remove_values on the `=` character to be consistent with the replace_values and new_values list?

In case the user specifies a command such as `o3de.py edit-gem-properties --remove-compatible-engines o3de=1.0`

Reply:
a valid version specifier looks like `o3de==1.0.0` or `o3de~=1.2.3`, a single `=` would not be valid